card/src/main/java/com/adyen/checkout/card/ui/AddressFormInput.kt

```diff
@@ -177,19 +189,25 @@ class AddressFormInput @JvmOverloads constructor(
                 @Suppress("UNUSED_VALUE")
                 isErrorFocused = true
                 textInputLayoutProvinceTerritory?.requestFocus()
+                textInputLayoutState?.requestFocus()
             }
             textInputLayoutProvinceTerritory?.showError(localizedContext.getString(provinceTerritoryValidation.reason))
+            textInputLayoutState?.showError(localizedContext.getString(provinceTerritoryValidation.reason))
         }
     }
 
     internal fun updateCountries(countryList: List<AddressListItem>) {
         countryAdapter.setItems(countryList)
         countryList.firstOrNull { it.selected }?.let {
             val selectedSpecification = AddressSpecification.fromString(it.code)
-            if (formContainer.childCount == 0) {
+            if (selectedSpecification != currentSpec) {
```

Comment:
Can you clean up this code a bit? We probably don't need the `run` block

Reply:
I guess we need the run block just for initial state when output has no countries.

Reply:
I'll check again if we can make it better.

Reply:
We don't need the null check here. At first when there's no selected country `AddressSpecification.fromString()` will return default. And then if there's a country it's gonna work as expected.